Source/WebKit/Shared/Cocoa/CoreIPCSecureCoding.mm

```diff
@@ -29,9 +29,60 @@
 #import "CoreIPCSecureCoding.h"
 
 #import "ArgumentCodersCocoa.h"
+#import "AuxiliaryProcessCreationParameters.h"
+#import "WKCrashReporter.h"
+#import <WebCore/RuntimeApplicationChecks.h>
+#import <wtf/text/StringHash.h>
 
 namespace WebKit {
 
+namespace SecureCoding {
+
+static std::unique_ptr<HashSet<String>>& internalClassNamesExemptFromSecureCodingCrash()
+{
+    static dispatch_once_t onceToken;
+    static NeverDestroyed<std::unique_ptr<HashSet<String>>> exemptClassNames;
+    dispatch_once(&onceToken, ^{
+        if (WebCore::isInAuxiliaryProcess())
+            return;
+
+        id object = [[NSUserDefaults standardUserDefaults] objectForKey:@"WebKitCrashOnSecureCodingWithExemptClassesKey"];
+        if (!object)
+            return;
+
+        exemptClassNames.get() = WTF::makeUnique<HashSet<String>>();
```

Comment:
Using std::optional would prevent an unnecessary allocation, but it would require more memory in the common case where this isn't used.  This is probably better.